app/scripts/components/common/cookie-consent/index.tsx

```diff
@@ -42,67 +66,104 @@ export const CookieConsent = ({
     )}; path=/; expires=${closeConsent ? '0' : setCookieExpiration()}`;
   };
 
+  const setSessionData = () => {
+    if (typeof window !== 'undefined') {
+      const checkForSessionDate = window.sessionStorage.getItem(SESSION_KEY);
+      if (!checkForSessionDate) {
+        window.sessionStorage.setItem(SESSION_KEY, 'true');
+      }
+    }
+  };
+  useEffect(() => {
+    if (sessionStart !== 'true' && !cookieConsentResponded) {
```

Comment:
I wonder if there is any way to simplify the logic so the flow can be more clear. 
ex. is there any case that sessionStart is not true (which I think it means that this component was never rendered before? - please correct me if I am wrong) but `cookieConsentReponded` is true?

Reply:
That is correct the only instance where `sessionStart` would be `false` is if the component was never rendered. Leading to a session token never being set. There is the potential that the `cookieConsentReponded` is true based on previous input in the form. But if they have already responded the component should not display in the browser.